R/RcppExports.R

```diff
@@ -1,53 +1,53 @@
+#' average number of contacts in group \eqn{i} reported by participants in
+#' group \eqn{j}.
+#' @param demography_vector Demography vector. Entry \eqn{pp_{i}} gives
+#' proportion of total population in group \eqn{i}
+#' (model will normalise if needed).
+#' @param susceptibility A matrix giving the susceptibility of individuals in
+#' demographic group \eqn{i} and risk group \eqn{j}.
+#' @param iterations Number of solver iterations. Defaults to 1,000.
+#' @param tolerance Solver error tolerance.
+#' @param step_rate The solver step rate. Defaults to 1.9 as a value found to
+#' work well.
+#' @param adapt_step Boolean, whether the solver step rate should be changed
+#' based on the solver error. Defaults to TRUE.
 #'
-#' @keywords epidemic model
-#' @export
-final_size_cpp <- function(r0, contact_matrix, demography_vector, prop_initial_infected, prop_suscep, iterations = 30L) {
-    .Call('_finalsize_final_size_cpp', PACKAGE = 'finalsize', r0, contact_matrix, demography_vector, prop_initial_infected, prop_suscep, iterations)
+#' @return A vector of final sizes, of the size (N demography groups *
+#' N risk groups).
+.solve_iterative <- function(contact_matrix, demography_vector, susceptibility, iterations = 10000L, tolerance = 1e-6, step_rate = 1.9, adapt_step = TRUE) {
+    .Call('_finalsize_solve_final_size_iterative', PACKAGE = 'finalsize', contact_matrix, demography_vector, susceptibility, iterations, tolerance, step_rate, adapt_step)
 }
 
-#' @title Calculate final epidemic size with risk groups using Eigen backend
-#'
-#' @description This function calculates final epidemic size using SIR model
-#' for a heterogeneously mixing population, using an Eigen backend.
+#' @title Newton solver for final size.
 #'
-#' # Solver options
-#'
-#' The `control` argument accepts a list of solver options, with the iterative
-#' solver taking two extra arguments than the Newton solver.
-#'
-#' ## Common options
+#' @param contact_matrix Social contact matrix. Entry \eqn{mm_{ij}} gives
+#' average number of contacts in group \eqn{i} reported by participants in
+#' group \eqn{j}.
+#' @param demography_vector Demography vector. Entry \eqn{pp_{i}} gives
+#' proportion of total population in group \eqn{i}
+#' (model will normalise if needed).
+#' @param susceptibility A matrix giving the susceptibility of individuals in
+#' demographic group \eqn{i} and risk group \eqn{j}.
+#' @param iterations Number of solver iterations. Defaults to 1,000.
+#' @param tolerance Solver error tolerance.
 #'
-#' 1. `iterations`: The number of iterations over which to solve for the final
-#' size, unless the error is below the solver tolerance.
-#' 2. `tolerance`: The solver tolerance, set to `1e-6` by default; solving for
-#' final size ends when the error drops below this tolerance.
+#' @return A vector final sizes, of the size (N demography groups *
+#' N risk groups).
+#' @return A two dimensional array of final sizes per age-risk group.
```

Comment:
Two `@return` fields

Reply:
Removed in https://github.com/epiverse-trace/finalsize/pull/73/commits/17e198a3cb896016840cbb759d9e0a03be6ae037